tests/basic/src/test/java/org/jboss/hal/testsuite/test/runtime/ServerPreviewTest.java

```diff
@@ -0,0 +1,74 @@
+package org.jboss.hal.testsuite.test.runtime;
+
+import java.io.IOException;
+import java.util.Map;
+
+import org.jboss.arquillian.core.api.annotation.Inject;
+import org.jboss.arquillian.drone.api.annotation.Drone;
+import org.jboss.arquillian.junit.Arquillian;
+import org.jboss.hal.meta.token.NameTokens;
+import org.jboss.hal.resources.Ids;
+import org.jboss.hal.testsuite.Console;
+import org.jboss.hal.testsuite.Random;
+import org.jboss.hal.testsuite.creaper.ManagementClientProvider;
+import org.jboss.hal.testsuite.fragment.AddResourceDialogFragment;
+import org.jboss.hal.testsuite.fragment.finder.FinderFragment;
+import org.jboss.hal.testsuite.fragment.finder.FinderPath;
+import org.jboss.hal.testsuite.util.ConfigUtils;
+import org.jboss.hal.testsuite.util.ServerEnvironmentUtils;
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.openqa.selenium.WebDriver;
+import org.wildfly.extras.creaper.core.online.OnlineManagementClient;
+
+@RunWith(Arquillian.class)
+public class ServerPreviewTest {
+
+    private static final OnlineManagementClient client = ManagementClientProvider.createOnlineManagementClient();
+    private static final ServerEnvironmentUtils serverEnvironmentUtils = new ServerEnvironmentUtils(client);
+    private static final String EDIT_URL_LABEL = "Edit URL";
+
+    @AfterClass
+    public static void tearDown() throws IOException {
+        client.close();
+    }
+
+    @Drone
+    private WebDriver browser;
+
+    @Inject
+    private Console console;
+
+    @Test
+    public void editUrl() throws IOException {
+        String url = Random.name();
+        FinderFragment fragment = goToEditModeAndGetFinder();
+        AddResourceDialogFragment editURLDialog = console.addResourceDialog();
+        editURLDialog.getForm().text("url", url);
+        editURLDialog.add();
+        Map<String, String> mainAttributes = fragment.preview().getMainAttributes();
+        Assert.assertTrue("\"URL\" field is present in Main Attributes", mainAttributes.containsKey("URL"));
+        Assert.assertEquals("\"URL\" field contains value set by test", mainAttributes.get("URL"), url);
```

Comment:
I'm missing the most important test of the value of the `href`link's attribute. I understand that this will require to add or change the UI abstraction of the preview.

Reply:
Moreover, it would be nice to add a test to verify the icon type (model vs. custom) - see description of the feature ... `The user should also be able to see whether the URL comes from the management model or whether it was specified by him. `